### PR TITLE
feat(dekuIntegration): add sign_payload and key

### PR DESCRIPTION
**Pull request by gsebil08**
# Context

For some messages, Deku needs to sign the payload

# Proposal

Make `msg_handler` returning a tuple `(payload, payload_signature)` of type `bytes * bytes option`

### CODE REVIEW

File: lib/pnode/failure_detector.ml [hunk 1]
```diff
@@ -91,6 +91,7 @@ let create_message node message (recipient : Peer.t) =
       sender = Client.address_of !node;
       recipient = recipient.address;
       payload = Encoding.pack bin_writer_message message;
+      payload_signature = None;
```
Comment (gsebil08) [inline lib/pnode/failure_detector.ml:94]: @rosalogia this is the actual only change on this PR.

File: lib/pnode/pnode.mli [hunk 2]
```diff
@@ -14,7 +14,7 @@ to initialize a server that runs asynchronously. Returns
 a reference to the node. *)
 val init :
   ?preprocess:(Message.t -> Message.t) ->
-  msg_handler:(Message.t -> bytes) ->
+  msg_handler:(Message.t -> bytes * bytes option) ->
```
Comment (gsebil08) [inline lib/pnode/pnode.mli:17]: I though a lot (too much) about this, I think making `msg_handler` returning a `bytes * bytes option` (where payload:bytes and payload_signature:bytes option) is the best option.
What do you think @rosalogia?

**Comment by rosalogia** [comment]
Can we merge #33 first? :sweat_smile: I don't mind being the one to rebase this PR after that.

**Comment by gsebil08** [comment]
>Can we merge https://github.com/marigold-dev/pollinate/pull/33 first? 😅 I don't mind being the one to rebase this PR after that.

Of course 😉

**Comment by gsebil08** [comment]
Closing in favor of #48

**Pull request closed without merging**